hassio/src/addon-view/config/hassio-addon-config.ts

```diff
@@ -29,35 +34,68 @@ import type { HomeAssistant } from "../../../../src/types";
 import { suggestAddonRestart } from "../../dialogs/suggestAddonRestart";
 import { hassioStyle } from "../../resources/hassio-style";
 
+const SUPPORTED_UI_TYPES = ["string", "select", "boolean", "integer", "float"];
+
 @customElement("hassio-addon-config")
 class HassioAddonConfig extends LitElement {
-  @property({ attribute: false }) public hass!: HomeAssistant;
-
   @property({ attribute: false }) public addon!: HassioAddonDetails;
 
-  @internalProperty() private _error?: string;
+  @property({ attribute: false }) public hass!: HomeAssistant;
 
   @property({ type: Boolean }) private _configHasChanged = false;
 
   @property({ type: Boolean }) private _valid = true;
 
-  @query("ha-yaml-editor", true) private _editor!: HaYamlEditor;
+  @internalProperty() private _canShowSchema = false;
+
+  @internalProperty() private _error?: string;
+
+  @internalProperty() private _options?: Record<string, unknown>;
+
+  @internalProperty() private _yamlMode = false;
+
+  @query("ha-yaml-editor") private _editor?: HaYamlEditor;
 
   protected render(): TemplateResult {
     return html`
       <h1>${this.addon.name}</h1>
-      <ha-card header="Configuration">
+      <ha-card>
+        <div class="header">
+          <h2>Configuration</h2>
+          <div class="card-menu">
+            <ha-button-menu corner="BOTTOM_START" @action=${this._handleAction}>
+              <mwc-icon-button slot="trigger">
+                <ha-svg-icon .path=${mdiDotsVertical}></ha-svg-icon>
+              </mwc-icon-button>
+              <mwc-list-item .disabled=${!this._canShowSchema}>
+                ${this._yamlMode ? "Edit in UI" : "Edit in YAML"}
+              </mwc-list-item>
+              <mwc-list-item class="warning">
+                Reset to defaults
+              </mwc-list-item>
+            </ha-button-menu>
+          </div>
+        </div>
+
         <div class="card-content">
-          <ha-yaml-editor
-            @value-changed=${this._configChanged}
-          ></ha-yaml-editor>
+          ${!this._yamlMode && this._canShowSchema && this.addon.schema
+            ? html`<ha-form
+                .data=${this._options!}
+                @value-changed=${this._configChanged}
+                .schema=${this.addon.schema}
+                .error=${this._error}
```

Comment:
This won't work, it would need to be a map just like data. But `this._error` doesn't seem to be set anywhere anyway 🤷‍♂️